### ISSUE TITLE
Definitions for overlapping namespaces are not merged

### ISSUE THREAD

**Issue by mvidner** [YorickPeterse/ruby-lint#181]
My team has a large project (https://github.com/yast/) consisting of many packages (rpms, gems).
One package, a base library, defines a `Yast` namespace, and I generated a definition file for it. 
Then in a different package there is a UI library in the `Yast::UI` namespace, and I generated definitions for that one too.
But I cannot make ruby-lint see both sets of definitions at once. They overwrite each other instead of getting merged.

A test case consisting of several small files is in https://gist.github.com/mvidner/c02b0367a622eddc373774f291a24b83

**Comment by mvidner** [comment]
Of the two approaches that I tried:

```sh
ruby-lint -c cfg-top.yml    app.rb
ruby-lint -c cfg-nested.yml app.rb
```

only the first one actually relies on definition merging.

The second one did not work because the `UI` in `Project` did not trigger a load of the `Project::UI` definition. I fix that in #184.
